src/py21cmmc/likelihood.py

```diff
@@ -682,6 +684,242 @@ def store(self, model, storage):
             storage.update({k + "_%s" % i: v for k, v in m.items()})
 
 
+class LikelihoodPlanckPowerSpectra(LikelihoodBase):
+    r"""A likelihood template to use Planck power spectrum.
+
+    It makes use of the clik code developed by the planck collaboration.
+    Go to the planck legacy archive website to download the code and data and get information: https://pla.esac.esa.int
+    You should download the "baseline" data package.
+    This likelihood currently supports Planck_lensing, Planck_highl_TTTEEE and Planck_lowl_EE data. Could easily be extended if required.
+
+    Parameters
+    ----------
+    datafolder : folder that contains Planck data if "clik" format.
+    name_lkl : str, default = Planck_lowl_EE, the planck likelihood to compute. choice: Planck_lensing, Planck_highl_TTTEEE, Planck_lowl_EE
+    """
+
+    required_cores = ((core.CoreLightConeModule, core.CoreCMB),)
+
+    def __init__(
+        self,
+        *args,
+        datafolder="blah",
+        name_lkl="Planck_lowl_EE",
+        A_planck_prior_center=1,
+        A_planck_prior_variance=0.1,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.path_clik = datafolder
+        self.name = name_lkl
+        self.A_planck_prior_center = A_planck_prior_center
+        self.A_planck_prior_variance = A_planck_prior_variance
+
+        if "Planck_lensing" in self.name:
+            self.lensing = True
+        else:
+            self.lensing = False
+        if "Planck_highl_TTTEEE" in self.name:
+            self.TTTEEE = True
+        else:
+            self.TTTEEE = False
+        if "Planck_lowl_EE" in self.name:
+            self.EE = True
+        else:
+            self.EE = False
+        if not self.TTTEEE and not self.EE and not self.lensing:
+            raise AttributeError(
+                "I did not understand name %s" % (self.name)
+                + " please choose between "
+                + "Planck_lensing, Planck_highl_TTTEEE, Planck_lowl_EE"
+            )
+
+        self.initialize = True
+        if self.initialize:
+            self.initialize_clik_and_class(datafolder, name_lkl)
+
+    def reduce_data(self, ctx):
+        """Get the CMB power spectra and the nuisance parameter A_planck from the coreCMB."""
+        cl = ctx.get("cl_cmb")
+        A_planck = ctx.get("A_planck", 1)
+        data = {"cl_cmb": cl, "A_planck_cmb": A_planck}
+
+        return data
+
+    def computeLikelihood(self, model):
+        """
+        Compute the likelihood.
+
+        This is the likelihood arising from Planck Lite (2018).
+
+        Parameters
+        ----------
+        cosmo : contains cosmological observables computed with CLASS
+            Exactly the output of CLASS
+
+        Returns
+        -------
+        lnl : float
+            The log-likelihood for the given model.
+        """
+        cl = model["cl_cmb"]
+
+        if self.lensing:
+            my_clik = my_clik_lensing
```

Comment:
This is super confusing because `my_clik_lensing` is a global. I think the reason for it being global is that it can't be pickled, so it can't be stored as part of the class. If this is the case, that's fine-ish (there are ways around it, but you then have to implemente `__getstate__` and `__setstate__` on the object, which is kind of annoying).  However, I'd at least add a comment like `# this is a global variable defined in initialize_clik_and_class`.

Reply:
This is a bit detailed about using clik. Do you mind commenting on this please @PoulinV ?

Reply:
I think this is what @PoulinV and I had discussed just after the Pisa workshop, and we came to the agreement that at least at first, the best way is to use it as a global. So that's fine, but it would be good to just have a comment over the variable telling us where it is defined.